File: 1013 - omaior/omaior.hs
import Prelude
import System.IO
import Text.Printf

maiordedois :: Int -> Int -> Int
maiordedois a b = do
    let sub = a - b
    let aff = abs sub
    let j = (a + b + aff)
    j `div` 2


maiordetres :: Int -> Int -> Int -> Int
maiordetres a b c = do
    let p1 = maiordedois a b
    maiordedois p1 c

    
main = do
    linha <- getLine
    let lista = words linha
        a = read( lista !! 0 ) :: Int
        b = read( lista !! 1 ) :: Int
        c = read( lista !! 2 ) :: Int

    let maior = maiordetres a b c
    printf "%d eh o maior\n" maior

File: 1013 - omaior/omaior.pas
program omaior(Input, Output);


function maiordedois( var a: integer; var b: integer) : integer;
begin    
    maiordedois := ((a + b + (abs(a - b))) div 2 );
end;

function maiordetres( var a : integer; var b : integer; var c : integer ) : integer;
var
    p1: integer;
begin
    p1 := maiordedois(a, b);
    maiordetres := maiordedois(p1, c);
end;

var
    a, b, c, maior : integer;

begin
    read(a, b, c);
    maior := maiordetres(a, b, c);
    Writeln( maior ,' eh o maior');
end.